automated install/basic-install.sh

```diff
@@ -435,30 +441,33 @@ update_repo() {
     # but since they are local, their scope does not go beyond this function
     # This helps prevent the wrong value from being assigned if you were to set the variable as a GLOBAL one
     local directory="${1}"
-    local curdir
+    local curBranch
 
     # A variable to store the message we want to display;
     # Again, it's useful to store these in variables in case we need to reuse or change the message;
     # we only need to make one change here
     local str="Update repo in ${1}"
-
-    # Make sure we know what directory we are in so we can move back into it
-    curdir="${PWD}"
-    # Move into the directory that was passed as an argument
-    cd "${directory}" &> /dev/null || return 1
+    # Move into the directory that was passed as an argument    
+    pushd "${directory}" &> /dev/null || return 1
     # Let the user know what's happening
     printf "  %b %s..." "${INFO}" "${str}"
     # Stash any local commits as they conflict with our working code
     git stash --all --quiet &> /dev/null || true # Okay for stash failure
     git clean --quiet --force -d || true # Okay for already clean directory
     # Pull the latest commits
     git pull --quiet &> /dev/null || return $?
+    # Check current branch. If it is master, then reset to the latest availible tag.
+    # In case extra commits have been added after tagging/release (i.e in case of metadata updates/README.MD tweaks)    
+    curBranch=$(git rev-parse --abbrev-ref HEAD)
+    if [[ "${curBranch}" == "master" ]]; then
+         git reset --hard "$(git describe --abbrev=0)" || return $?
```

Comment:
Isn't it dangerous to put whatever `git describe` returns into this call without checking if the returned value is something we want?

I played around and saw this:
```
root@munichpi:~/FTL# git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
root@munichpi:~/FTL# git pull
Already up-to-date.
root@munichpi:~/FTL# git describe --abbrev=0
fatal: No annotated tags can describe 'b60d63f448179cb139755eb8dcfd9d5335df3fd4'.
However, there were unannotated tags: try --tags.
root@munichpi:~/FTL# git describe --abbrev=0 --tags
v4.3.1
```

Reply:
So you think we should add the `--tags` to the end?

Reply:
Looking at it, adding --tags is at least safer in terms of output. We shouldn't ever be in a situation on `master` (the only branch this code will check) where there are no tags for `describe` to... describe